Spain population by provinces chart

JTrillo opened the pull request:
I'm going to sumarize changes:
- I had to add two dependencies (**d3-composite-projections** & **topojson-client**) and one development dependency (**@types/topojson-client**). That's why **package.json** and **package-lock.json** are modified.
- I modified a little bit the pod **main-chart-list** to show new chart. Just that.
- I use the file **pods/spain-population-chart/spain-population-chart.config.ts** to set chart styles and other stuff like svg dimensions.
- We could create a folder to store all json files like **spain-province.json** because I think we are going to use them more than once.

JTrillo commented:
I forgot to adjoint the source of the data: https://www.ine.es/jaxiT3/Tabla.htm?t=2852

brauliodiez commented on pods/main-chart-list/main-chart.vm.ts:
```diff
@@ -11,3 +11,10 @@ export const createDefaultSampleChart = (): ChartInfoVm => ({
+
+export const createSpainPopulationChart = (): ChartInfoVm => ({
+  title: "Spain population by provinces",
+  description: "Spain population by provinces (data updated on 1st January 2020)",
```
We have to include the source where we extracted the data

JTrillo replied:
True. I'm gonna fix that quickly.

JTrillo replied:
I'm thinking we should create a new field on the interface ***ChartInfoVm*** (**pods/main-chart-list/main-chart.vm.ts**) called *source*, puttin the source outside *description* field. Then we could use that new filed on component **chart-info-card.tsx** as a link and not like plain text (or also plain text but separeted from description).  What do you think about it?

I adjoint the current state of **chart-info-card.tsx**:
![image](https://user-images.githubusercontent.com/33926302/85553301-1a078600-b624-11ea-9aed-7706dec4888c.png)

brauliodiez replied:
Source and Tags as well (tag array) we need it to get good SEO support

JTrillo replied:
Something like this (?):
![image](https://user-images.githubusercontent.com/33926302/85568941-6bb70d00-b632-11ea-93d6-72c4ea068ecf.png)

brauliodiez replied:
That's great, I would add this as an issue and take it in a separate branch

brauliodiez commented on pods/spain-population-chart/spain-population-chart.config.ts:
```diff
@@ -0,0 +1,42 @@
+  strokeWidth: 2,
+  stroke: "#bc5b40",
+  fill: "#f88f70",
+};
```
We can make use of classes, check the approach that is already in master if we rename the css file to .module.css we get automatically support of css modules

JTrillo replied:
Something like this (?): 
![image](https://user-images.githubusercontent.com/33926302/85568805-4924f400-b632-11ea-8286-9f9abbd993ae.png)

JTrillo commented:
Ok Braulio I think you can review it again :)

JTrillo commented:
Major changes:
- Created component **common-app/chart-info.component.tsx**. This component can be used inside chart component file. The new component add below the header the long description, the source and tags.
- Renamed page from english to spanish. Now file is **pages/espana-poblacion-provincias-2020.tsx**. **ChartInfoComponent** is used here.
- Short description is used on chart info cards.
- Added long description to my chart. Also created a table next to the chart on **pods/spain-population-chart/spain-population-chart.component.tsx**.

JTrillo commented:
If you have any doubt please tell me.

brauliodiez commented:
About the legend it should be displayed where the mouse is located (a litte bit up where the mouse point is located) and move along.

brauliodiez commented:
Remove the paging on the table and add scrolling,and for instance add sorting per column, so you can decide to sort by Province or by population (if it already provides that MUI table and is easy to add it).

On the other hand population should be measured in something "Habitantes"

brauliodiez commented:
About the tags, they should have a different background color

brauliodiez commented:
I will open a separate case to provide support to markdown to the descriptions (no hurry with that but will make sense look ,more professional)

brauliodiez commented:
Design should be responsive if you make smaller the page (e.g. mobile), spain map should take all the space and the table falldown.

JTrillo commented:
Thank you for the feedback Braulio. I will make the changes during next week.

JTrillo commented:
OK Braulio I think you can review the branch again. I modified what you said to me on comments (except provide support to markdown to the descriptions).

brauliodiez commented on pods/spain-population-chart/components/table.component.tsx:
```diff
@@ -0,0 +1,123 @@
+    return a[1] - b[1];
+  });
+  return stabilizedThis.map((el) => el[0]);
+}
```
Should we move this to common/sort or common/utils ?
